src/disassembler.rs

```diff
@@ -0,0 +1,17 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt;
+
+// Supported architectures
+enum Architecture {
+    X86,
+    X86_64,
+}
+
```

Comment:
Good point. This will depend mostly on whether we want to differentiate `X86` and `X86_64` or not. If we want, we'll need some extra info from Capstone disassembler.